entity-framework/core/testing/index.md

```diff
@@ -1,117 +1,116 @@
 ---
-title: Testing code that uses EF Core - EF Core
+title: Testing applications that use EF Core - EF Core
 description: Different approaches to testing applications that use Entity Framework Core
-author: ajcvickers
-ms.date: 04/22/2020
+author: roji
+ms.date: 11/07/2021
 uid: core/testing/index
 ---
-# Testing code that uses EF Core
+# Testing EF Core Applications
 
-Testing code that accesses a database requires either:
+Testing is an important concern to almost all application types - testing allows you to be sure your application work correctly, and makes it instantly known if its behavior regresses in the future. Since testing may affect how your code is architected, it's highly recommended to plan for a testing early, and to ensure good coverage as your application evolves. This section provides an overview of various testing strategies for applications using EF Core, shows how each one can be implemented and discusses best practices.
 
-* Running queries and updates against the same database system used in production.
-* Running queries and updates against some other easier to manage database system.
-* Using test doubles or some other mechanism to avoid using a database at all.
+## Unit and integration testing
 
-This document outlines the trade-offs involved in each of these choices and shows how EF Core can be used with each approach.
+Although there are many kinds of testing, in this section we'll concentrate on two types of automated testing: unit and integration testing. Without going into formal definitions, unit tests are about isolating a single component and testing against it, with all related components (or dependencies) stubbed out. Importantly, unit tests never interact with a real database, but instead replace the database with a [test double](https://martinfowler.com/bliki/TestDouble.html) which fulfills the same contracts. Integration tests, in contrast, test actual components working together - including your production database system.
```

Comment:
I predict you're going to have a fun time arguing with people who disagree with these statements. Again, I wonder if we should be making these distinctions, as opposed to just showing techniques for different types of tests.

Reply:
Interesting - these seem like pretty uncontroversial definitions to me... and I even explicitly say we're avoiding formal definitions (and link to Martin Fowler, who specifically talks about replacing real databases)... Do you think it's controversial to say that integration tests include your real database, whereas unit tests do not?

The thing is that within these docs, it's important to make the distinction between these two testing types, since the whole docs are split along those lines. I also think lots of users are a bit blurry on what kind of testing they want (and also what it is they're trying to test), so making this clearer may help.

Reply:
I've tweaked things a bit around this. I'm still using *integration* and *unit* tests in the same way (i.e. with or without the production database system), but I introduce these terms as definitions for simplicity's sake in the new overview page. Hope that helps.

Reply:
Hi @roji,

I also use the phase _automated testing_ when talking about testing EF Core code. The problem with the term _integration testing_ is people thinks that hard when in fact using EF Core is much easier that mocking or subbing a repository.